### PR TITLE
(Feature) Percentage Spread

### LINKED ISSUE #409: (Feature) Spread percentage in orderbook
Show spread percentage next to the spread amount
image

### PR DESCRIPTION
**Pull request by Agupane**
Closes #409 

Before:
image
After:
image

### CODE REVIEW

**Comment by fvictorio** [comment]
This swaps the "My Size" and "Price" columns, since that's the order in the design. It also makes more sense with how the spread is shown.

File: src/components/erc20/marketplace/order_book.tsx [hunk 1]
```diff
@@ -132,7 +144,7 @@ class OrderBookTable extends React.Component<Props> {
                                     borderTop: true,
                                 }}
                             >
-                                {spread.toFixed(UI_DECIMALS_DISPLAYED_PRICE_ETH)}
+                                {percentageSpread.toFixed(0)}%
```
Comment (fvictorio) [inline src/components/erc20/marketplace/order_book.tsx]: Req: I think 0 decimal places won't make sense most of the time. I think 2 should work for the time being.

File: src/store/selectors.ts [hunk 2]
```diff
@@ -134,17 +134,32 @@ export const getSpread = createSelector(
...
+    },
+);
+
 export const getOrderBook = createSelector(
     getOpenSellOrders,
     getOpenBuyOrders,
     getMySizeOrders,
     getSpread,
-    (sellOrders, buyOrders, mySizeOrders, spread): OrderBook => {
+    getSpreadInPercentage,
```
Comment (fvictorio) [inline src/store/selectors.ts]: Req: These two selectors are not being used.

File: src/store/selectors.ts [hunk 3]
```diff
@@ -134,17 +134,32 @@ export const getSpread = createSelector(
...
+    getOpenSellOrders,
+    (buyOrders, sellOrders) => {
+        if (!buyOrders.length || !sellOrders.length) {
+            return new BigNumber(0);
+        }
+
+        const lowestPriceSell = sellOrders[sellOrders.length - 1].price;
+        const highestPriceBuy = buyOrders[0].price;
+        const spread = lowestPriceSell.minus(highestPriceBuy);
+        return spread.dividedBy(lowestPriceSell).multipliedBy(100);
```
Comment (fvictorio) [inline src/store/selectors.ts]: PP: This has a lot of duplication with `getSpread`. Maybe we can compute the percentage from the `getSpread` and `getOpenSellOrders` selectors? Or just make one selector that returns a pair?

File: src/components/erc20/marketplace/order_book.tsx (hunk 1 above)
Comment (Agupane) [reply src/components/erc20/marketplace/order_book.tsx]: Done

File: src/store/selectors.ts (hunk 2 above)
Comment (Agupane) [reply src/store/selectors.ts]: I'd removed them, thanks!

File: src/store/selectors.ts (hunk 3 above)
Comment (Agupane) [reply src/store/selectors.ts]: I'd used the `getSpread` and `getOpenSellOrders`, please check it again. I think making one selector that returns a pair it's not too easy to read as having just two selectors for different things

**Review by unjapones — APPROVED** [review]

**Review by mariano-aguero — APPROVED** [review]
LGTM, tested on ganache

**Pull request merged**